### PR TITLE
Fix codegen for multi-reg EH vars

### LINKED ISSUE #41232: EH WriteThru breaks on Arm (in Reflection.Emit.ILGenerator.Emit)
Isolated repo, see CI tests https://github.com/dotnet/runtime/pull/41227

Does seem very constant failures on arm; all in `ILGenerator.Emit` on lots of different tests (though mainly Regex); though various different error message

```
      Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
         at System.Reflection.Emit.ILGenerator.Emit(System.Reflection.Emit.OpCode, System.Reflection.Emit.LocalBuilder)
         at System.Text.RegularExpressions.RegexCompiler.Ldloc(System.Reflection.Emit.LocalBuilder)
         at System.Text.RegularExpressions.RegexCompiler.GenerateOneCode()
         at System.Text.RegularExpressions.RegexCompiler.GenerateForwardSection()
         at System.Text.RegularExpressions.RegexCompiler.GenerateGo()
         at System.Text.RegularExpressions.RegexLWCGCompiler.FactoryInstanceFromCode(System.String, System.Text.RegularExpressions.RegexCode, System.Text.RegularExpressions.RegexOptions, Boolean)
         at System.Text.RegularExpressions.RegexCompiler.Compile(System.String, System.Text.RegularExpressions.RegexCode, System.Text.RegularExpressions.RegexOptions, Boolean)
         at System.Text.RegularExpressions.Regex.Compile(System.String, System.Text.RegularExpressions.RegexCode, System.Text.RegularExpressions.RegexOptions, Boolean)
         at System.Text.RegularExpressions.Regex..ctor(System.String, System.Text.RegularExpressions.RegexOptions, System.TimeSpan, System.Globalization.CultureInfo)
         at System.Text.RegularExpressions.Regex..ctor(System.String, System.Text.RegularExpressions.RegexOptions)
         at MethodImplOptionsTests.RunMethodImplOptionsTest(System.String, System.String, System.String, System.String, System.String)
         at MethodImplOptionsTests.Main()
```
```
System.TypeInitializationException : The type initializer for 'System.Data.Common.DbConnectionOptions' threw an exception.
--- System.ArgumentNullException : Value cannot be null. (Parameter 'local')

 at System.Data.Common.DbConnectionOptions.IsKeyNameValid(String keyname) in /_/src/libraries/Common/src/System/Data/Common/DbConnectionOptions.Common.cs:line 420
   at System.Data.Common.DbConnectionOptions.ParseInternal(Dictionary`2 parsetable, String connectionString, Boolean buildChain, Dictionary`2 synonyms, Boolean firstKey) in /_/src/libraries/Common/src/System/Data/Common/DbConnectionOptions.Common.cs:line 568
   at System.Data.Common.DbConnectionOptions..ctor(String connectionString, Dictionary`2 synonyms, Boolean useOdbcRules) in /_/src/libraries/System.Data.Odbc/src/Common/System/Data/Common/DbConnectionOptions.cs:line 53
   at System.Data.Odbc.OdbcConnectionString..ctor(String connectionString, Boolean validate) in /_/src/libraries/System.Data.Odbc/src/System/Data/Odbc/OdbcConnectionString.cs:line 17
   at System.Data.Odbc.OdbcConnectionFactory.CreateConnectionOptions(String connectionString, DbConnectionOptions previous) in /_/src/libraries/System.Data.Odbc/src/System/Data/Odbc/OdbcConnectionFactory.cs:line 40
   at System.Data.ProviderBase.DbConnectionFactory.GetConnectionPoolGroup(DbConnectionPoolKey key, DbConnectionPoolGroupOptions poolOptions, DbConnectionOptions& userConnectionOptions) in /_/src/libraries/Common/src/System/Data/ProviderBase/DbConnectionFactory.cs:line 198
   at System.Data.Odbc.OdbcConnection.ConnectionString_Set(DbConnectionPoolKey key) in /_/src/libraries/System.Data.Odbc/src/System/Data/Odbc/OdbcConnectionHelper.cs:line 89
   at System.Data.Odbc.OdbcConnection.ConnectionString_Set(String value) in /_/src/libraries/System.Data.Odbc/src/System/Data/Odbc/OdbcConnectionHelper.cs:line 83
   at System.Data.Odbc.OdbcConnection.set_ConnectionString(String value) in /_/src/libraries/System.Data.Odbc/src/System/Data/Odbc/OdbcConnection.cs:line 59
   at System.Data.Odbc.OdbcConnection..ctor(String connectionString) in /_/src/libraries/System.Data.Odbc/src/System/Data/Odbc/OdbcConnection.cs:line 26
   at System.Data.Odbc.Tests.IntegrationTestBase..ctor() in /_/src/libraries/System.Data.Odbc/tests/IntegrationTestBase.cs:line 16
   at System.Data.Odbc.Tests.ReaderTests..ctor()
----- Inner Stack Trace -----
   at System.Reflection.Emit.ILGenerator.Emit(OpCode opcode, LocalBuilder local) in /_/src/coreclr/src/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.cs:line 821
   at System.Text.RegularExpressions.RegexCompiler.Ldloc(LocalBuilder lt) in /_/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs:line 352
   at System.Text.RegularExpressions.RegexCompiler.PushTrack(LocalBuilder lt) in /_/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs:line 609
   at System.Text.RegularExpressions.RegexCompiler.GenerateOneCode() in /_/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs:line 4053
   at System.Text.RegularExpressions.RegexCompiler.GenerateForwardSection() in /_/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs:line 902
   at System.Text.RegularExpressions.RegexCompiler.GenerateGo() in /_/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs:line 3326
   at System.Text.RegularExpressions.RegexLWCGCompiler.FactoryInstanceFromCode(String pattern, RegexCode code, RegexOptions options, Boolean hasTimeout) in /_/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexLWCGCompiler.cs:line 61
   at System.Text.RegularExpressions.RegexCompiler.Compile(String pattern, RegexCode code, RegexOptions options, Boolean hasTimeout) in /_/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs:line 146
   at System.Text.RegularExpressions.Regex.Compile(String pattern, RegexCode code, RegexOptions options, Boolean hasTimeout) in /_/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex
```

category:correctness
theme:register-allocator
skill-level:expert
cost:medium

### PR DESCRIPTION
**Pull request by CarolEidt**
Multi-reg EH vars were not properly being stored at each def.
Fix the dumping of multi-reg nodes with register assignments.
Temporarily enable EHWriteThru to get CI testing with it on.

Fix #41232

### CODE REVIEW

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib PTAL
I will re-disable EHWriteThru and re-run CI before merging.

File: src/coreclr/src/jit/gentree.cpp [hunk 1]
```diff
@@ -10499,73 +10499,94 @@ void Compiler::gtDispNode(GenTree* tree, IndentStack* indentStack, __in __in_z _
     }
 }
 
+#if FEATURE_MULTIREG_RET
+//----------------------------------------------------------------------------------
+// gitDispRegCount: store multi-reg value to a local
```
Comment (BruceForstall) [inline src/coreclr/src/jit/gentree.cpp:10504]: nit: typo: gitDispRegCount

Comment (BruceForstall) [inline src/coreclr/src/jit/gentree.cpp:10504]: "store multi-reg value to a local" -> copy/paste error?

File: src/coreclr/src/jit/gentree.cpp [hunk 2]
```diff
@@ -10499,73 +10499,94 @@ void Compiler::gtDispNode(GenTree* tree, IndentStack* indentStack, __in __in_z _
...
+//
+unsigned Compiler::gtDispRegCount(GenTree* tree)
+{
+    if (tree->IsCopyOrReload())
+    {
+        // GetRegCount() will return only the number of valid regs for COPY or RELOAD,
+        // but we want to print all positions, so we get the reg count for op1.
+        return gtDispRegCount(tree->gtGetOp1());
+    }
+    if (!tree->IsMultiRegNode())
```
Comment (BruceForstall) [inline src/coreclr/src/jit/gentree.cpp:10526]: For consistency, shouldn't this and the next 'if' also be 'else if', like the ones that follow?

**Review by BruceForstall — APPROVED** [review]

**Pull request merged**